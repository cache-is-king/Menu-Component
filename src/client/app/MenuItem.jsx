import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/MenuItemStyles.css';

const MenuItem = (props) => {
  const { item } = props;
  return (
    <div className={styles.menuItemBox}>
      <span className={styles.itemName}>{item.foodItem}</span>
      <span className={styles.itemCost}>{item.cost}</span>
    </div>
  );
};

MenuItem.propTypes = {
  item: PropTypes.shape({
    foodItem: PropTypes.string.isRequired,
    cost: PropTypes.number.isRequired,
    tags: PropTypes.string.isRequired,
  }).isRequired,
};

export default MenuItem;
