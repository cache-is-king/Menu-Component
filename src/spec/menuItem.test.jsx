import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuItem from '../client/app/MenuItem';
import fakeMenus from './fakeData';

Enzyme.configure({ adapter: new Adapter() });

describe('MenuItem Component', () => {
  let component = '';
  test('should render single Menu Item component correctly', () => {
    component = Enzyme.shallow(<MenuItem item={fakeMenus.lunch[0]} />);
    expect(component).toMatchSnapshot();
  });
});
