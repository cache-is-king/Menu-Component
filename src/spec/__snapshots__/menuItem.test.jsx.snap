// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Component should render single Menu Item component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuItem
    item={
      Object {
        "cost": 99,
        "foodItem": "sandwich",
        "tags": "none",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className={undefined}
        >
          sandwich
        </span>,
        <span
          className={undefined}
        >
          99
        </span>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "sandwich",
          "className": undefined,
        },
        "ref": null,
        "rendered": "sandwich",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 99,
          "className": undefined,
        },
        "ref": null,
        "rendered": 99,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className={undefined}
          >
            sandwich
          </span>,
          <span
            className={undefined}
          >
            99
          </span>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "sandwich",
            "className": undefined,
          },
          "ref": null,
          "rendered": "sandwich",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 99,
            "className": undefined,
          },
          "ref": null,
          "rendered": 99,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
